### PR TITLE
expose the encryption level of packets

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -320,11 +320,15 @@ func (s *session) handlePacketImpl(p *receivedPacket) error {
...
-	if utils.Debug() {
-		utils.Debugf("<- Reading packet 0x%x (%d bytes) for connection %x @ %s", hdr.PacketNumber, len(data)+len(hdr.Raw), hdr.ConnectionID, time.Now().Format("15:04:05.000"))
-	}
 
 	packet, err := s.unpacker.Unpack(hdr.Raw, hdr, data)
+	if utils.Debug() {
+		if err != nil {
+			utils.Debugf("<- Reading packet 0x%x (%d bytes) for connection %x @ %s", hdr.PacketNumber, len(data)+len(hdr.Raw), hdr.ConnectionID, time.Now().Format("15:04:05.000"))
+		} else {
+			utils.Debugf("<- Reading packet 0x%x (%d bytes) for connection %x, %s @ %s", hdr.PacketNumber, len(data)+len(hdr.Raw), hdr.ConnectionID, packet.encryptionLevel, time.Now().Format("15:04:05.000"))
```
Comment (lucas-clemente) [inline session.go]: I just realized that we are logging time now – we shouldn't do that here, that's the logger's job. I've opened a new issue for that #444.

File: handshake/crypto_setup_client.go [hunk 2]
```diff
@@ -276,27 +275,31 @@ func (h *cryptoSetupClient) validateVersionList(verTags []byte) bool {
...
-			return nil, err
+			return nil, protocol.EncryptionUnspecified, err
 		}
 	}
-
-	return (&crypto.NullAEAD{}).Open(dst, src, packetNumber, associatedData)
+	nullAEAD := &crypto.NullAEAD{}
+	res, err := nullAEAD.Open(dst, src, packetNumber, associatedData)
+	if err != nil {
+		return res, protocol.EncryptionUnspecified, err
```
Comment (marten-seemann) [inline handshake/crypto_setup_client.go]: How could I miss that? 👎

File: handshake/crypto_setup_client.go [hunk 3]
```diff
@@ -276,27 +275,31 @@ func (h *cryptoSetupClient) validateVersionList(verTags []byte) bool {
...
+	}
+	return res, protocol.EncryptionUnencrypted, err
```
Comment (marten-seemann) [inline handshake/crypto_setup_client.go]: `return res, protocol.EncryptionUnencrypted, nil`

**Comment by marten-seemann** [comment]
I rebased the branch and fixed all the issues while doing so.

**Pull request merged**